File: frontend/src/mocks/models/Book.js
import { getMockLibraries } from "mocks/models/Library";

export const getMockBooks = (titles) => {
  return titles.map((title, index) => {
    return {
      id: index,
      author: `author-${index}`,
      genre: `genre-${index}`,
      language: `language-${index}`,
      library: getMockLibraries(["MockLibrary"]).reduce((a, v) => ({
        ...a,
        [v]: v,
      })),
      pages: index + 10,
      plot: `plot-${index}`,
      remoteId: index,
      title: title,
      year: 1720 + index,
      cover:
        "https://bibliotecadigitale.unicam.it/Library/DIR.CIV.-IV-B-601/DIR.CIV.-IV-B-601_0000.JPG",
    };
  });
};

File: backend/src/main/kotlin/it/unicam/cs/digital_library/model/Book.kt
package it.unicam.cs.digital_library.model

import org.hibernate.annotations.OnDelete
import org.hibernate.annotations.OnDeleteAction
import javax.persistence.*

@Entity
@Table(uniqueConstraints = [UniqueConstraint(columnNames = ["library_id", "remoteId"])])
data class Book(
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    val id: Long = 0,
    val title: String,
    val author: String,
    val pages: Int,
    @ManyToOne
    @OnDelete(action = OnDeleteAction.CASCADE)
    @JoinColumn(name = "library_id")
    val library: Library,
    val remoteId: Long,
    // optional
    val language: String?,
    val year: Int?,
    val genre: String?,
    val plot: String?
)